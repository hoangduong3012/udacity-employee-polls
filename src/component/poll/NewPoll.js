import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import { useState } from "react";
import { handleAddQuestion } from "../../actions/questions";
import NavigationBar from "../common/NavigationBar";
import { Footer } from "../common/Footer";
import { Container } from "react-bootstrap";

const NewPoll = (props) => {
  const navigate = useNavigate();
  const [optionOne, setOptionOne] = useState("");
  const [optionTwo, setOptionTwo] = useState("");

  const handleSubmit = (e) => {
    e.preventDefault();
    props.dispatch(handleAddQuestion(optionOne, optionTwo));
    navigate("/");
  };
  return (
    <div>
      <NavigationBar />
      <Container className="text-center p-5 new-container">
        <h2>Would you rather?</h2>
        <form className="form" onSubmit={handleSubmit}>
          <div className="mb-3">
            <label htmlFor="formGroupExampleInput" className="form-label">
              Poll option One
            </label>
            <input
              type="text"
              className="form-control"
              id="optionOne"
              data-testid="optionOne"
              name="optionOne"
              value={optionOne}
              onChange={(e) => setOptionOne(e.target.value)}
              placeholder="Option One"
              required
            />
          </div>
          <div className="mb-3">
            <label htmlFor="formGroupExampleInput2" className="form-label">
              Poll option Two
            </label>
            <input
              type="text"
              className="form-control"
              data-testid="optionTwo"
              id="optionTwo"
              name="optionTwo"
              value={optionTwo}
              onChange={(e) => setOptionTwo(e.target.value)}
              placeholder="Option Two"
              required
            />
          </div>
          <input type="submit" value="submit" data-testid="submitButton" />
        </form>
      </Container>
      <Footer />
    </div>
  );
};

const mapStateToProps = ({ dispatch }) => ({
  dispatch,
});

export default connect(mapStateToProps)(NewPoll);
