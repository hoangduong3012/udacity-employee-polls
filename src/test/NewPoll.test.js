import NewPoll from "../component/poll/NewPoll";
import React from "react";
import { Provider } from "react-redux";
import { store } from "../store";
import { BrowserRouter as Router } from "react-router-dom";
import { render, fireEvent, getByTestId } from '@testing-library/react';

describe("App", () => {
  it("will match snapshot", () => {
    var component = render(
      <Provider store={store}>
        <Router>
        <NewPoll />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
