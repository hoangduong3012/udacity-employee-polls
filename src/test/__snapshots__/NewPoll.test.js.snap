// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="bgg navbar navbar-expand-sm navbar-dark"
        >
          <button
            aria-controls="navbarScroll"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            data-bs-target="#navbarScroll"
            data-bs-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <span
            class="p-3 navbar-brand"
          >
            Web Poll App
          </span>
          <div
            class="navbar-collapse collapse"
            id="navbarScroll"
          >
            <div
              class="container"
            >
              <div
                class="navbar-nav"
              >
                <a
                  class="nav-link"
                  href="/"
                >
                  Home
                </a>
                <a
                  class="nav-link"
                  href="/leaderboard"
                >
                  Leaderboard
                </a>
                <a
                  class="nav-link"
                  href="/add"
                >
                  New
                </a>
                <div
                  class="justify-content-end navbar-collapse collapse"
                >
                  <span
                    class="p-2 navbar-text"
                  >
                    Signed in as: 
                    <img
                      alt="Avatar"
                      class="rounded-circle p-2"
                      height="37"
                      loading="lazy"
                      src="https://mdbcdn.b-cdn.net/img/Photos/Avatars/img (31).webp"
                    />
                  </span>
                  <span
                    class="navbar-text"
                  >
                    <button
                      class="btn btn-outline-warning btn-sm"
                      data-testid="logout-button"
                      type="button"
                    >
                      <span
                        class="ml-3"
                      >
                        signout
                      </span>
                    </button>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </nav>
        <div
          class="text-center p-5 new-container container"
        >
          <h2>
            Would you rather?
          </h2>
          <form
            class="form"
          >
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="formGroupExampleInput"
              >
                Poll option One
              </label>
              <input
                class="form-control"
                data-testid="optionOne"
                id="optionOne"
                name="optionOne"
                placeholder="Option One"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                for="formGroupExampleInput2"
              >
                Poll option Two
              </label>
              <input
                class="form-control"
                data-testid="optionTwo"
                id="optionTwo"
                name="optionTwo"
                placeholder="Option Two"
                required=""
                type="text"
                value=""
              />
            </div>
            <input
              data-testid="submitButton"
              type="submit"
              value="submit"
            />
          </form>
        </div>
        <div>
          <footer
            class="bgg"
          >
            <div
              class="footer-contents container"
            >
              <div>
                <a
                  class="childs nav-link"
                  href="/"
                >
                  Home
                </a>
                <a
                  class="childs nav-link"
                  href="/leaderboard"
                >
                  Leaderboard
                </a>
                <a
                  class="childs nav-link"
                  href="/addPoll"
                >
                  New
                </a>
                <a
                  class="childs nav-link"
                  href="/FAQs"
                >
                  FAQs
                </a>
                <a
                  class="childs nav-link"
                  href="/addPoll"
                >
                  About
                </a>
              </div>
              <hr />
              <div>
                © 2024 Hoang Bui Company, Inc
              </div>
            </div>
          </footer>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="bgg navbar navbar-expand-sm navbar-dark"
      >
        <button
          aria-controls="navbarScroll"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          data-bs-target="#navbarScroll"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <span
          class="p-3 navbar-brand"
        >
          Web Poll App
        </span>
        <div
          class="navbar-collapse collapse"
          id="navbarScroll"
        >
          <div
            class="container"
          >
            <div
              class="navbar-nav"
            >
              <a
                class="nav-link"
                href="/"
              >
                Home
              </a>
              <a
                class="nav-link"
                href="/leaderboard"
              >
                Leaderboard
              </a>
              <a
                class="nav-link"
                href="/add"
              >
                New
              </a>
              <div
                class="justify-content-end navbar-collapse collapse"
              >
                <span
                  class="p-2 navbar-text"
                >
                  Signed in as: 
                  <img
                    alt="Avatar"
                    class="rounded-circle p-2"
                    height="37"
                    loading="lazy"
                    src="https://mdbcdn.b-cdn.net/img/Photos/Avatars/img (31).webp"
                  />
                </span>
                <span
                  class="navbar-text"
                >
                  <button
                    class="btn btn-outline-warning btn-sm"
                    data-testid="logout-button"
                    type="button"
                  >
                    <span
                      class="ml-3"
                    >
                      signout
                    </span>
                  </button>
                </span>
              </div>
            </div>
          </div>
        </div>
      </nav>
      <div
        class="text-center p-5 new-container container"
      >
        <h2>
          Would you rather?
        </h2>
        <form
          class="form"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="formGroupExampleInput"
            >
              Poll option One
            </label>
            <input
              class="form-control"
              data-testid="optionOne"
              id="optionOne"
              name="optionOne"
              placeholder="Option One"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="formGroupExampleInput2"
            >
              Poll option Two
            </label>
            <input
              class="form-control"
              data-testid="optionTwo"
              id="optionTwo"
              name="optionTwo"
              placeholder="Option Two"
              required=""
              type="text"
              value=""
            />
          </div>
          <input
            data-testid="submitButton"
            type="submit"
            value="submit"
          />
        </form>
      </div>
      <div>
        <footer
          class="bgg"
        >
          <div
            class="footer-contents container"
          >
            <div>
              <a
                class="childs nav-link"
                href="/"
              >
                Home
              </a>
              <a
                class="childs nav-link"
                href="/leaderboard"
              >
                Leaderboard
              </a>
              <a
                class="childs nav-link"
                href="/addPoll"
              >
                New
              </a>
              <a
                class="childs nav-link"
                href="/FAQs"
              >
                FAQs
              </a>
              <a
                class="childs nav-link"
                href="/addPoll"
              >
                About
              </a>
            </div>
            <hr />
            <div>
              © 2024 Hoang Bui Company, Inc
            </div>
          </div>
        </footer>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
